virtcontainers/store/filesystem.go

```diff
@@ -0,0 +1,310 @@
+var ConfigStoragePath = filepath.Join("/var/lib", StoragePathSuffix, SandboxPathSuffix)
+
+// RunStoragePath is the sandbox runtime directory.
+// It will contain one state.json and one lock file for each created sandbox.
+var RunStoragePath = filepath.Join("/run", StoragePathSuffix, SandboxPathSuffix)
+
+// RunVMStoragePath is the vm directory.
+// It will contain all guest vm sockets and shared mountpoints.
+var RunVMStoragePath = filepath.Join("/run", StoragePathSuffix, VMPathSuffix)
+
+func itemToFile(item Item) (string, error) {
+	switch item {
+	case Configuration:
+		return ConfigurationFile, nil
+	case State:
+		return StateFile, nil
+	case Network:
+		return NetworkFile, nil
+	case Hypervisor:
+		return HypervisorFile, nil
+	case Agent:
+		return AgentFile, nil
+	case Process:
+		return ProcessFile, nil
+	case Lock:
+		return LockFile, nil
+	case Mounts:
+		return MountsFile, nil
+	case Devices, DeviceIDs:
+		return DevicesFile, nil
+	}
+
+	return "", fmt.Errorf("Unknown item %s", item)
+}
+
+type filesystem struct {
+	ctx context.Context
+
+	path    string
+	rawPath string
+
+	lockTokens map[string]*os.File
+}
+
+// Logger returns a logrus logger appropriate for logging Store filesystem messages
+func (f *filesystem) logger() *logrus.Entry {
+	return storeLog.WithFields(logrus.Fields{
+		"subsystem": "store",
+		"backend":   "filesystem",
+		"path":      f.path,
+	})
+}
+
+func (f *filesystem) trace(name string) (opentracing.Span, context.Context) {
+	if f.ctx == nil {
+		f.logger().WithField("type", "bug").Error("trace called before context set")
+		f.ctx = context.Background()
+	}
+
+	span, ctx := opentracing.StartSpanFromContext(f.ctx, name)
+
+	span.SetTag("subsystem", "store")
+	span.SetTag("type", "filesystem")
+	span.SetTag("path", f.path)
+
+	return span, ctx
+}
+
+func (f *filesystem) itemToPath(item Item) (string, error) {
+	fileName, err := itemToFile(item)
+	if err != nil {
+		return "", err
+	}
+
+	return filepath.Join(f.path, fileName), nil
+}
+
+func (f *filesystem) initialize() error {
+	_, err := os.Stat(f.path)
+	if err == nil {
+		return nil
+	}
+
+	// Our root path does not exist, we need to create the initial layout:
+	// The root directory, a lock file and a raw files directory.
+
+	// Root directory
+	if err := os.MkdirAll(f.path, DirMode); err != nil {
+		return err
+	}
+
+	// Raw directory
+	if err := os.MkdirAll(f.rawPath, DirMode); err != nil {
+		return err
+	}
+
+	// Lock
+	lockPath := filepath.Join(f.path, LockFile)
+
+	lockFile, err := os.Create(lockPath)
+	if err != nil {
+		f.delete()
+		return err
+	}
+	lockFile.Close()
+
+	return nil
+}
+
+func (f *filesystem) new(ctx context.Context, path string, host string) error {
+	f.ctx = ctx
+	f.path = path
+	f.rawPath = filepath.Join(f.path, "raw")
+	f.lockTokens = make(map[string]*os.File)
+
+	f.logger().Debugf("New filesystem store backend for %s", path)
+
+	return f.initialize()
+}
+
+func (f *filesystem) delete() {
+	os.RemoveAll(f.path)
```

Comment:
This call return an error, so why not return it?

Reply:
Done.